src/utils/validations/validator/validator.js

```diff
@@ -0,0 +1,36 @@
+const isLegacy = (func) => { return typeof func === 'object'; };
+
+const validator = validationFunctions => (value, props) => (
+  new Promise(async (resolve, reject) => {
+    let results = [];
+
+    if (Array.isArray(validationFunctions)) {
+      results = validationFunctions.map((func) => {
+        if (isLegacy(func)) {
+          const valid = func.validate(value, props);
+          if (valid) {
+            return Promise.resolve();
+          }
+          return Promise.reject(func.message());
+        }
+        return func(value, props);
+      });
```

Comment:
I wonder if we can do it more readable and separate this function.

Reply:
Yeah agreed, I've stripped out half of the code and I'll separate some of it into a new function for the legacy, I'll push up the changes soon